### ISSUE TITLE
improve kube-derive

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#167]
One thing I frequently find myself doing is shorthand creating objects:

```rust
let foo = Foo::new("myname").within("namespace").spec(FooSpec::new(...))
// post foo to Api
```

Before we could possible implement these on `Object<FooSpec, K>`
, but now it's kind of hard since we don't enforce a lot of structure and only got the `k8s_openapi::{Resource,Metadata}` traits to work with.

Maybe we should define our own `KubeExt` trait to provide some helpers on objects created by `kube-derive`. Maybe we could also implement this trait for `Metadata` implementors in a limited way?

**Comment by clux** [comment]
Adding a generic constructor with a generic trait is non-trivial - at least without bringing back Object and forcing kube-derive depending on kube (which is undesirable as long as kube pulls in an http client).

But have done an easy impl that reaches into the Resource trait, and also uses the arbitrary type name (given to the `#[derive(CustomResource)]`) to construct a `Foo::new(name: &str, spec: FooSpec)`.

**Comment by clux** [comment]
Released in 0.28.1. That fixes the heart of the problem (generated type not needing `Default`).
